cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -185,8 +185,9 @@ func DefaultedStaticInitConfiguration() (*kubeadmapi.InitConfiguration, error) {
 		LocalAPIEndpoint: kubeadmapiv1.APIEndpoint{AdvertiseAddress: "1.2.3.4"},
 		BootstrapTokens:  []kubeadmapiv1.BootstrapToken{PlaceholderToken},
 		NodeRegistration: kubeadmapiv1.NodeRegistrationOptions{
-			CRISocket: kubeadmconstants.DefaultDockerCRISocket, // avoid CRI detection
-			Name:      "node",
+			CRISocket:       kubeadmconstants.DefaultDockerCRISocket, // avoid CRI detection
+			Name:            "node",
+			ImagePullPolicy: v1.PullIfNotPresent,
```

Comment:
note that fields like tokens, endpoints, node name, etc and others are different.
they have dynamic defaulting i.e. they are obtained by checking the node on runtime.

ImagePullPolicy has a static default. if empty it is always v1.PullIfNotPresent.
so the correct place to do this is in:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/defaults.go

Reply:
and if we are going to update defaults.go we need to remove these lines:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go#L835-L838

Reply:
Thanks！Sorry. Maybe my description is not clear.  we know kubeadm will give us the message as the following when we run kubeadm config print init-defaults|:
`apiVersion: kubeadm.k8s.io/v1beta3                                                                                                                                                                                                         
bootstrapTokens:
- groups:
  - system:bootstrappers:kubeadm:default-node-token
  token: abcdef.0123456789abcdef
  ttl: 24h0m0s
  usages:
  - signing
  - authentication
........`
This PR is try to add ImagePullPolicy field to kubeadm's output message when we run the above command.
I think that "....v1beta3/defaults.go" is only used to set  default values of fields. Is it right?

Reply:
Thanks!  Maybe my description is not clear. we know that kubeadm will give us  yaml message as below when we run  "kubeadm config print init-defaults" or  "kubeadm config print join-defaults":
`apiVersion: kubeadm.k8s.io/v1beta3
bootstrapTokens:
- groups:
  - system:bootstrappers:kubeadm:default-node-token
  token: abcdef.0123456789abcdef
  ttl: 24h0m0s
  usages:
  - signing
  - authentication
kind: InitConfiguration
localAPIEndpoint:
  advertiseAddress: 1.2.3.4
  bindPort: 6443
nodeRegistration:
.........`
This PR be  try to add "imagePullPolicy" field to yaml message like above when we run the commands.
I have built a "kubeadm" binary base on master code. I can not find the "imagePullPolicy" field when I run "kubeadm config print init-defaults" ,even if the default value of imagePullPolicy is "".
I think that "defaults.go" is only used to set default values of fileds. Is it right?